src/main/java/com/transferwise/openbanking/client/oauth/PrivateKeyJwtAuthentication.java

```diff
@@ -35,7 +35,7 @@ public void addClientAuthentication(GetAccessTokenRequest getAccessTokenRequest,
 
     private String getClientAssertionToken(AspspDetails aspspDetails) {
         JwtClaims jwtClaims = new JwtClaims();
-        jwtClaims.setIssuer(aspspDetails.getClientId());
+        jwtClaims.setIssuer(aspspDetails.getIssuer() !=null ? aspspDetails.getIssuer() : aspspDetails.getClientId());
         jwtClaims.setSubject(aspspDetails.getClientId());
         jwtClaims.setAudience(aspspDetails.getPrivateKeyJwtAuthenticationAudience());
```

Comment:
note: in https://github.com/transferwise/confirmation-of-payee/pull/167 we'll wire up that `PrivateKeyJwtAuthenticationAudience` so it can return something else besides the clientId